### DISCUSSION TITLE
Parallel Routes default.js vs catchall

### DISCUSSION THREAD

**Discussion by Brendonovich** [vercel/next.js#67724] [App Router]
We're implementing parallel routes in Solid Router and I'm curious why Next recommends the default.js file instead of [...catchall].js for unmatched slots. As far as I can tell they act the same but I might be missing something. Would anyone have insight into why this choice was made?

**Comment by morgs32** [comment]
You ever get an answer to this @Brendonovich ?

**Reply by Brendonovich** [reply]
Something I didn't account for when I asked this question was that a route is considered 'matched' as long as one of any slot is matched in it - I thought that at least the `children` slot had to match.
Say you've got a route with slots A and B.
If slot A has `default.js`, then that's what will be shown when it isn't matched but slot B is. If neither are matched then `not-found.js` will be shown.
If slot A has a catchall instead, then slot A will _always_ be matched. Its main content will be shown if that matches, else the catchall will be shown, and your `not-found.js` will _never_ be reached, since at least one slot is being matched.
Might be wrong but I that's how I understand it now.

**Reply by morgs32** [reply]
Ahh well assuming that's how it works you explained it terrifically. Thanks
